packages/http-client-csharp/generator/Microsoft.TypeSpec.Generator/src/PostProcessing/GeneratedCodeWorkspace.cs

```diff
@@ -260,5 +265,34 @@ public async Task PostProcessAsync()
                     break;
             }
         }
+
+        internal static async Task<Compilation?> LoadBaselineContract()
+        {
+            string fullPath;
+            string ns = CodeModelGenerator.Instance.TypeFactory.PrimaryNamespace;
+            string projectFilePath = Path.GetFullPath(Path.Combine(CodeModelGenerator.Instance.Configuration.ProjectDirectory, $"{ns}.csproj"));
+            if (!File.Exists(projectFilePath))
+                return null;
+
+            var baselineVersion = ProjectRootElement.Open(projectFilePath).Properties.SingleOrDefault(p => p.Name == ApiCompatPropertyName)?.Value;
+            if (baselineVersion != null)
+            {
+                var nugetGlobalPackageFolder = SettingsUtility.GetGlobalPackagesFolder(Settings.LoadDefaultSettings(projectFilePath));
+                var nugetFolder = Path.Combine(nugetGlobalPackageFolder, ns.ToLowerInvariant(), baselineVersion, "lib", LastContractAssemblyNetVersion);
+                fullPath = Path.Combine(nugetFolder, $"{ns}.dll");
+                if (File.Exists(fullPath))
+                {
+                    return await CreateLastContractFromDll(Path.Combine(nugetFolder, $"{ns}.xml"), fullPath);
+                }
+                else
+                {
+                    CodeModelGenerator.Instance.Emitter.ReportDiagnostic(
+                        DiagnosticCodes.BaselineContractMissing,
+                        $"Can't find Baseline contract assembly ({ns}@{baselineVersion}) from Nuget Global Package Folder at {fullPath}. " +
+                        $"Please make sure the baseline nuget package has been installed properly");
```

Comment:
How would the baseline package be installed? I think in the azure-sdk-for-net repo we have build magic to get the baseline contract. I'm wondering if we should pull down the nuget package from the generator if we don't find it.